### PR TITLE
B1 m2 5069

### PR DESCRIPTION
**Pull request by prushin**
Add LndLockedError and NetworkException errors and handlers.

### CODE REVIEW

File: src/LndGrpc/Client.hs [hunk 1]
```diff
@@ -80,11 +84,22 @@ runStreamServer rpc env req handler = do
...
           (setupGrpcClient $ envLndConfig env)
           close
           (\grpc -> rawStreamServer rpc grpc () req $ const handler)
+
+catchExceptions :: MonadUnliftIO m => m (Either LndError a) -> m (Either LndError a)
+catchExceptions x =
+  x
+    `catches` [ Handler (\(e :: BlockedIndefinitelyOnMVar) -> pure (Left $ LndGrpcException $ inspect e)),
+                Handler
+                  ( \(ex :: IOException) -> case stripPrefix ("Network.Socket.connect" :: Text) (pack $ Prelude.show ex) of
```
Comment (tim2CF) [inline src/LndGrpc/Client.hs:101]: Should it be IOException? Maybe it's too wide + you need to parse text which is not ideal. Test this, maybe such handler will be enough:

https://github.com/21it/src/blob/a22ac3f061d88025ef6c56a6e47f5b3ec37d8228/bitfinex-client/src/BitfinexClient/Rpc/Generic.hs#L138-L151

File: src/LndGrpc/Client.hs [hunk 2]
```diff
@@ -32,11 +34,14 @@ runUnary ::
...
-    liftIO (Right <$> this)
-      `catch` (\(e :: BlockedIndefinitelyOnMVar) -> pure $ Left e)
+    catchExceptions $ liftIO (Right <$> this)
   pure $ case res of
     Right (Right (Right (Right (_, _, Right x)))) ->
       Right x
+    Right (Right (Right (Right (_, _, Left "wallet not created, create one to enable full RPC access")))) ->
+      Left $ LndLockedError "wallet not created"
+    Right (Right (Right (Right (_, _, Left "wallet locked, unlock it to enable full RPC access")))) ->
+      Left $ LndLockedError "wallet locked"
```
Comment (tim2CF) [inline src/LndGrpc/Client.hs:44]: Let's have 2 separate constructors for Locked (non fatal error) and NonExistent (fatal) wallet errors.

**Review by tim2CF — CHANGES_REQUESTED** [review]

File: src/LndGrpc/Client.hs (hunk 2 above)
Comment (prushin) [reply src/LndGrpc/Client.hs]: Done

File: src/LndGrpc/Client.hs (hunk 1 above)
Comment (prushin) [reply src/LndGrpc/Client.hs:102]: It catches IOException with type NoSuchThing, Looks like HttpException is not thrown by runUnary and cannot be catched.

Comment (prushin) [reply src/LndGrpc/Client.hs:102]: Done

**Pull request merged**